### PR TITLE
add inspect to returned data

### PR DESCRIPTION
**Pull request by anandkumarpatel**
send inspect data on change so api does not have to inspect anymore :)

### CODE REVIEW

**Comment by anandkumarpatel** [comment]
@podviaznikov

File: lib/publisher.js [hunk 1]
```diff
@@ -16,27 +15,26 @@ function RedisPublisherStream() {
...
-      catch (err) {
-        callback(err);
-      }
-    }
-    else {
-      callback();
+    if (!chunk) { return callback(); }
+    var ev;
+    try {
+      ev = JSON.parse(chunk.toString());
```
Comment (podviaznikov) [inline lib/publisher.js:21]: nice!

**Comment by podviaznikov** [comment]
@anandkumarpatel tests are failing

**Comment by anandkumarpatel** [comment]
@podviaznikov yep forgot to add a file, should pass now

File: lib/publisher.js (hunk 1 above)
Comment (anandkumarpatel) [inline lib/publisher.js:21]: actually is there a reason you do this here? chunk is already be an object? was there cases where it was not?

**Comment by anandkumarpatel** [comment]
PING @podviaznikov

File: lib/publisher.js (hunk 1 above)
Comment (podviaznikov) [inline lib/publisher.js:21]: oh, than it's bug I guess. 
If there is no comment why I convert to string and than parse it. I'll look into it now.

Comment (anandkumarpatel) [inline lib/publisher.js:21]: one reason I see people do this is to make a copy of the object, but I do not think that is needed here

Comment (podviaznikov) [inline lib/publisher.js:21]: I think it should be string. Streams pass strings around, not objects (by default).

File: lib/publisher.js [hunk 2]
```diff
@@ -16,27 +15,26 @@ function RedisPublisherStream() {
...
+    } catch (err) {
+      return callback(err);
     }
+
+    datadog.inc(ev);
+    events.enhance(ev, function(err, event) {
+      var channel = process.env.DOCKER_EVENTS_NAMESPACE + event.status;
```
Comment (podviaznikov) [inline lib/publisher.js]: you don't handle `err` here

File: lib/publisher.js (hunk 1 above)
Comment (anandkumarpatel) [inline lib/publisher.js:21]: ok, so docker stream send chunk as string and internally we send object to write. got it, will leave. (we could optimize by only parsing if chunk == string)

File: lib/publisher.js (hunk 2 above)
Comment (anandkumarpatel) [inline lib/publisher.js]: good catch!

File: lib/publisher.js (hunk 1 above)
Comment (podviaznikov) [inline lib/publisher.js:21]: actually, seems that streams pass buffers and not strings. So we call .toString on buffer I believe.

File: lib/publisher.js [hunk 3]
```diff
@@ -16,27 +15,26 @@ function RedisPublisherStream() {
...
+      redis.publish(channel, JSON.stringify(event), function (err) {
+        if (err) {
+          debug('error publishing to redis');
+          error.log(err);
+          callback(err);
```
Comment (podviaznikov) [inline lib/publisher.js]: you need `return` here. Otherwise callback will be called twice.

File: lib/publisher.js [hunk 4]
```diff
@@ -16,27 +15,26 @@ function RedisPublisherStream() {
...
+    if (!chunk) { return callback(); }
+    var ev;
+    try {
+      ev = JSON.parse(chunk.toString());
+    } catch (err) {
+      return callback(err);
     }
+
+    datadog.inc(ev);
+    events.enhance(ev, function(err, event) {
```
Comment (podviaznikov) [inline lib/publisher.js]: we probably may rename `event` to `ev` or `e`. Text editors usually highlight `event` as keyword.

**Comment by anandkumarpatel** [comment]
responded to your comments @podviaznikov

**Comment by podviaznikov** [comment]
@anandkumarpatel other than typo in var name +1. No more comments

**Comment by anandkumarpatel** [comment]
cool @podviaznikov 
fixed type, hit the merge button if you approve :1234: :cactus: :facepunch: :smile: :cool: :computer:

**Pull request merged**